CHANGELOG.md

```diff
@@ -11,45 +11,118 @@ We use *breaking* word for marking changes that are not backward compatible (rel
+  New Store flags:
+  
+    * `--store.grpc.series-sample-limit` limits the amount of samples that might be retrieved on a single Series() call. By default it is 0. Consider enabling it by setting it to more than 0 if you are running on limited resources.
+    * `--store.grpc.series-max-concurrency` limits the number of concurrent Series() calls in Thanos Store. By default it is 20. Considering making it lower or bigger depending on the scale of your deployment.
+
+  New Store metrics:
+  
+    * `thanos_bucket_store_queries_dropped_total` shows how many queries were dropped due to the samples limit;
+    * `thanos_bucket_store_queries_concurrent_max` is a constant metric which shows how many Series() calls can concurrently be executed by Thanos Store;
+    * `thanos_bucket_store_queries_in_flight` shows how many queries are currently "in flight" i.e. they are being executed;
+    * `thanos_bucket_store_gate_duration_seconds` shows how many seconds it took for queries to pass through the gate in both cases - when that fails and when it does not.
+    
+  New Store tracing span:
+    * `store_query_gate_ismyturn` shows how long it took for a query to pass (or not) through the gate.
+    
+- [#1016](https://github.com/improbable-eng/thanos/pull/1016) Added option for another DNS resolver (miekg/dns client). 
+This to have SRV resolution working on [Golang 1.11+ with KubeDNS below v1.14](https://github.com/golang/go/issues/27546)
 
-New metrics:
-* `thanos_bucket_store_queries_dropped_total` shows how many queries were dropped due to the samples limit;
-* `thanos_bucket_store_queries_concurrent_max` is a constant metric which shows how many Series() calls can concurrently be executed by Thanos Store;
-* `thanos_bucket_store_queries_in_flight` shows how many queries are currently "in flight" i.e. they are being executed;
-* `thanos_bucket_store_gate_duration_seconds` shows how many seconds it took for queries to pass through the gate in both cases - when that fails and when it does not.
+   New Querier and Ruler flag: `-- store.sd-dns-resolver` which allows to specify resolver to use. Either `golang` or `miekgdns`
+   
+- [#986](https://github.com/improbable-eng/thanos/pull/986) Allow to save some startup & sync time in store gateway as it is no longer is needed to compute index-cache from block index on its own. 
+  For compatibility store GW still can do it, but it first checks bucket if there is index-cached uploaded already. 
+  In the same time, compactor precomputes the index cache file on every compaction. 
 
-New tracing span:
-* `store_query_gate_ismyturn` shows how long it took for a query to pass (or not) through the gate.
+  New Compactor flag: `--index.generate-missing-cache-file` was added to allow quicker addition of index cache files, if enabled it precomputes missing files on compactor startup. Note that it will take time and it's only one-off step per bucket.
```

Comment:
```suggestion
  New Compactor flag: `--index.generate-missing-cache-file` was added to allow quicker addition of index cache files. If enabled it precomputes missing files on compactor startup. Note that it will take time and it's only one-off step per bucket.
```